### PR TITLE
[IMP] estate: add constraints and further refine UI T#71276

### PR DESCRIPTION
**Pull request by ralberrto**
- Add required constraints to models
- Add constraints to validate offers expected and selling prices
- Add list of EstateProperties associated to PropertyTypes
- Add statusbar view to EstateProperty form
- Add db order to models
- Add sequence handler widget to PropertyTypes view
- Prevent creation/edition of PropertyTypes add color to PropetyTags
- Make the buttons in header vanish if property is sold or canceled
- Use attrs to hide fields, buttons, and make them readonly
- Make list views editable in-place
- Add decorations to entries based on fields' values
- Apply the **Available** filter associated to EstateProperty by default
- Add filter domain to the living_area field
- Add related field to PropertyOffer model
- Add computed field offer_count to PropertyType
- Add action and smart button linked to it filtering the view

### CODE REVIEW

**Comment by ralberrto** [comment]
Hey @edy1192 this _PR_ is ready. Could you please check it out? Thank you!

File: estate/views/estate_property_views.xml [hunk 1]
```diff
@@ -55,13 +78,16 @@
                                 <field name="facades" />
                                 <field name="garage" />
                                 <field name="garden" />
-                                <field name="garden_area" />
-                                <field name="garden_orientation" />
+                                <field attrs="{'invisible': [('garden', '=', False)]}" name="garden_area" />
+                                <field attrs="{'invisible': [('garden', '=', False)]}" name="garden_orientation" />
```
Comment (edy1192) [inline estate/views/estate_property_views.xml:82]: First is the field name then attributes, options...

```suggestion
                                <field name="garden_area" attrs="{'invisible': [('garden', '=', False)]}" />
                                <field name="garden_orientation" attrs="{'invisible': [('garden', '=', False)]}" />
```

File: estate/views/estate_property_offer_views.xml [hunk 2]
```diff
@@ -5,14 +5,26 @@
...
+                    string="Accept"
+                    type="object"
+                />
+                <button
+                    attrs="{'invisible': ['|', ('status', '=', 'accepted'), ('status', '=', 'refused')]}"
+                    icon="fa-times"
+                    name="action_set_status_refused"
+                    string="Refuse"
+                    type="object"
+                />
```
Comment (edy1192) [inline estate/views/estate_property_offer_views.xml:26]: `attrs` is the last option that we add. 

```suggestion
                <button
                    icon="fa-check"
                    name="action_set_status_accepted"
                    string="Accept"
                    type="object"
                    attrs="{'invisible': ['|', ('status', '=', 'accepted'), ('status', '=', 'refused')]}"
                />
                <button
                    icon="fa-times"
                    name="action_set_status_refused"
                    string="Refuse"
                    type="object"
                    attrs="{'invisible': ['|', ('status', '=', 'accepted'), ('status', '=', 'refused')]}"
                />
```

File: estate/views/estate_property_views.xml [hunk 3]
```diff
@@ -5,16 +5,21 @@
...
                 <field name="postcode" />
                 <field name="bedrooms" />
                 <field name="living_area" />
                 <field name="expected_price" />
                 <field name="selling_price" />
-                <field name="date_availability" />
+                <field name="date_availability" optional="hide" />
                 <field name="property_type_id" />
                 <field name="tag_ids" widget="many2many_tags" />
+                <field name="status" optional="hide" />
```
Comment (edy1192) [inline estate/views/estate_property_views.xml:22]: Why optional hide? I did not see an exercise that says that.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by ralberrto** [comment]
Hi @edy1192 I just finished making the necessary changes. Can you please check them out? Thank you!

File: estate/models/estate_property.py [hunk 4]
```diff
@@ -62,6 +64,15 @@ def _compute_best_price(self):
             else:
                 record.best_price = 0
 
+    @api.constrains("expected_price", "selling_price")
+    def _check_property_selling_and_expected_prices(self):
+        for record in self:
+            if (
+                float_utils.float_compare(record.selling_price, record.expected_price * 0.90, precision_rounding=2)
+                == -1
+            ) and record.selling_price != 0:
```
[the remark this answers is not in the document]
Comment (edy1192) [reply estate/models/estate_property.py]: Got it.

File: estate/views/estate_property_type_views.xml [hunk 5]
```diff
@@ -17,9 +18,30 @@
...
             <form>
                 <sheet>
+                    <div class="oe_stat_button" name="button_box">
+                        <button
+                            name="estate.estate_property_offer_action"
+                            class="oe_stat_button"
+                            icon="fa-money"
+                            type="action"
+                        >
+                            <field name="offer_ids" widget="statinfo" string="Offers" />
```
Comment (edy1192) [inline estate/views/estate_property_type_views.xml:28]: This should be the `offer_count` field.
```suggestion
                            <field name="offer_count" widget="statinfo" string="Offers" />
```

File: estate/models/estate_property_type.py [hunk 6]
```diff
@@ -1,8 +1,26 @@
...
+    @api.depends("offer_ids")
+    def _compute_offer_count(self):
+        for record in self:
+            record.offer_count = len(record.offer_ids)
+
+    _sql_constraints = [
+        (
+            "property_type_name_unique",
+            "UNIQUE (name)",
+            "A property type with the same name already exists",
```
[the remark this answers is not in the document]
Comment (edy1192) [reply estate/models/estate_property_type.py:19]: Pending.

File: estate/models/estate_property_type.py [hunk 7]
```diff
@@ -1,8 +1,26 @@
...
+    offer_count = fields.Integer(compute="_compute_offer_count")
+
+    @api.depends("offer_ids")
+    def _compute_offer_count(self):
+        for record in self:
+            record.offer_count = len(record.offer_ids)
+
+    _sql_constraints = [
+        (
+            "property_type_name_unique",
```
[the remark this answers is not in the document]
Comment (edy1192) [reply estate/models/estate_property_type.py:17]: Pending.

File: estate/models/estate_property_offer.py [hunk 8]
```diff
@@ -17,6 +18,7 @@ class EstatePropertyOffer(models.Model):
     property_id = fields.Many2one("estate.property", required=True)
     validity = fields.Integer(default=7, string="Validity (days)")
     date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline", string="Deadline")
+    property_type_id = fields.Many2one(related="property_id.property_type_id")
```
[the remark this answers is not in the document]
Comment (edy1192) [reply estate/models/estate_property_offer.py:22]: Pending.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by ralberrto** [comment]
Hey @edy1192 the changes have been completed. Would you please check them out?

**Review by edy1192 — APPROVED** [review]

**Pull request merged**